src/core/skills/characters/yijiang2014/qiangzhi.ts

```diff
@@ -0,0 +1,146 @@
+
+@CommonSkill({ name: 'qiangzhi', description: 'qiangzhi_description' })
+export class QiangZhi extends TriggerSkill {
+  isTriggerable(event: ServerEventFinder<GameEventIdentifiers.PhaseStageChangeEvent>, stage?: AllStage) {
+    return stage === PhaseStageChangeStage.StageChanged;
+  }
+
+  canUse(room: Room, owner: Player, content: ServerEventFinder<GameEventIdentifiers.PhaseStageChangeEvent>) {
+    return content.playerId === owner.Id && content.toStage === PlayerPhaseStages.PlayCardStageStart;
+  }
+
+  async onTrigger() {
+    return true;
+  }
+
+  async onEffect(room: Room, content: ServerEventFinder<GameEventIdentifiers.SkillEffectEvent>) {
+    const ownerId = content.fromId;
+
+    room.notify(
+      GameEventIdentifiers.AskForChoosingPlayerEvent,
+      EventPacker.createUncancellableEvent<GameEventIdentifiers.AskForChoosingPlayerEvent>({
+        toId: ownerId,
+        players: room.AlivePlayers.filter(
+          player => player.getCardIds(PlayerCardsArea.HandArea).length > 0 && player.Id !== ownerId,
+        ).map(player => player.Id),
+        requiredAmount: 1,
+        conversation: TranslationPack.translationJsonPatcher(
+          '{0}: please choose a player to show a hand card',
+          this.GeneralName,
+        ).extract(),
+      }),
+      ownerId,
+    );
+
+    const { selectedPlayers } = await room.onReceivingAsyncResponseFrom(
+      GameEventIdentifiers.AskForChoosingPlayerEvent,
+      ownerId,
+    );
+
+    if (selectedPlayers === undefined) {
+      return true;
+    }
+
+    room.notify(
+      GameEventIdentifiers.AskForChoosingCardFromPlayerEvent,
+      EventPacker.createUncancellableEvent<GameEventIdentifiers.AskForChoosingCardFromPlayerEvent>({
+        fromId: ownerId,
+        toId: selectedPlayers[0],
+        options: {
+          [PlayerCardsArea.HandArea]: room.getPlayerById(selectedPlayers[0]).getCardIds(PlayerCardsArea.HandArea)
+            .length,
+        },
+      }),
+      ownerId,
+    );
+
+    // just for interactive
+    await room.onReceivingAsyncResponseFrom(GameEventIdentifiers.AskForChoosingCardFromPlayerEvent, ownerId);
+
+    const candidateCard = room.getPlayerById(selectedPlayers[0]).getCardIds(PlayerCardsArea.HandArea);
+    const selectedCard = candidateCard[Math.floor(Math.random() * candidateCard.length)];
+
+    room.broadcast(GameEventIdentifiers.CardDisplayEvent, {
+      displayCards: [selectedCard],
+      translationsMessage: TranslationPack.translationJsonPatcher(
+        '{0} display {1} hand card {2}',
+        TranslationPack.patchPlayerInTranslation(room.getPlayerById(ownerId)),
+        TranslationPack.patchPlayerInTranslation(room.getPlayerById(selectedPlayers[0])),
+        TranslationPack.patchCardInTranslation(selectedCard),
+      ).extract(),
+    });
+
+    room.getPlayerById(ownerId).setFlag(this.GeneralName, Sanguosha.getCardById(selectedCard).BaseType);
+
+    return true;
+  }
+}
+
+@ShadowSkill
```

Comment:
clear flags in `whenRefresh` or `beforeUse`;

Reply:
Refresh skill at PlayCardStage, and remove this flag when refresh skill.
Add contriant: Can't trigger skill in other player's phase.